### PR TITLE
Migrate ExtendedFragmentFactoryTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate ExtendedFragmentFactoryTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt [hunk 1]
```diff
@@ -13,91 +13,61 @@
...
-        final ExtendedFragmentFactory testFF = new TestFragmentFactoryTest(baseFF);
-
-        testFF.instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME);
-
-        verify(baseFF, times(1)).instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME);
+    fun willCallBaseFactory() {
+        val baseFF = Mockito.mock(FragmentFactory::class.java)
+        val testFF: ExtendedFragmentFactory = TestFragmentFactoryTest(baseFF)
+        testFF.instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME)
+        Mockito.verify(baseFF, Mockito.times(1)).instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt:38]: Import `Mockito`

File: AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt [hunk 2]
```diff
@@ -13,91 +13,61 @@
...
-        verify(fragmentManager, times(1)).setFragmentFactory(testFF);
-
-        testFF.instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME);
-
-        verify(baseFactory, times(1)).instantiate(fakeClassLoader, MockFragmentClassLoader.FAKE_CLASS_NAME);
+    fun testAttachToActivity() {
+        val activity = Mockito.mock(AppCompatActivity::class.java)
+        val fragmentManager = Mockito.mock(FragmentManager::class.java)
+        val baseFactory = Mockito.mock(FragmentFactory::class.java)
+        Mockito.`when`(activity.supportFragmentManager).thenReturn(fragmentManager)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt:46]: cleanup: `when`

File: AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt [hunk 3]
```diff
@@ -13,91 +13,61 @@
...
+        Mockito.`when`(fragmentManager.fragmentFactory).thenReturn(baseFactory)
+        val testFF: ExtendedFragmentFactory = TestFragmentFactoryTest()
+        val result = testFF.attachToActivity<ExtendedFragmentFactory>(activity)
+        Assert.assertEquals(testFF, result)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ExtendedFragmentFactoryTest.kt:50]: import: `Assert`

**Review by david-allison — CHANGES_REQUESTED** [review]
Needs line spacing fixing

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
great

**Pull request merged**